x/nftmarket/keeper/nft_listing.go

```diff
@@ -58,6 +59,53 @@ func getTimeKey(prefix string, timestamp time.Time) []byte {
 	return bz
 }
 
+func (k Keeper) SaveNftListing(ctx sdk.Context, listing types.NftListing) {
+	k.SetNftListing(ctx, listing)
+	k.UpdateClass(ctx, listing)
+}
+
+func (k Keeper) UpdateClass(ctx sdk.Context, listing types.NftListing) {
+	switch listing.State {
+	case types.ListingState_LISTING:
+		k.SetClass(ctx, listing)
+	case types.ListingState_SUCCESSFUL_BID:
+		// todo delete nftid from class
+	}
+}
+
+func (k Keeper) SetClass(ctx sdk.Context, listing types.NftListing) {
+	store := ctx.KVStore(k.storeKey)
+	bzIdlist := store.Get(types.ClassKey(listing.ClassIdBytes()))
+	if bzIdlist == nil {
+		bz := k.cdc.MustMarshal(
+			&types.Class{
+				ClassId: listing.NftId.ClassId,
+				NftIds:  []string{listing.NftId.NftId},
+			},
+		)
+		store.Set(types.ClassKey(listing.ClassIdBytes()), bz)
+	} else {
+
+		// todo delete dumplicate nftid
+		class := types.Class{}
+		k.cdc.MustUnmarshal(bzIdlist, &class)
+		class.NftIds = append(class.NftIds, listing.NftId.NftId)
+		bz := k.cdc.MustMarshal(&class)
+		store.Set(types.ClassKey(listing.ClassIdBytes()), bz)
+	}
+}
+
+func (k Keeper) GetClass(ctx sdk.Context, classIdByte []byte) (types.Class, error) {
```

Comment:
Can you add what GetClass function does? And I think for class and history related operations, it would be good to be put on a separate file rather than keeping on nft_listing.go file.

Reply:
got it

Reply:
I renamed GetClass to GetListedClassByClassIdBytes.
Did you need comment?